pkg/certificate/signer.go

```diff
@@ -0,0 +1,528 @@
+func (s *signerImpl) startPeriodicCACheck(namespace string) {
+	if s.caTicker != nil {
+		return
+	}
+
+	s.caTicker = time.NewTicker(CACheckInterval)
+	s.caStopCh = make(chan struct{})
+
+	go func() {
+		for {
+			select {
+			case <-s.caTicker.C:
+				logger.V(1).Info("Performing periodic CA check", "namespace", namespace)
+
+				// Check if CA secret exists first
+				caSecretClient := s.dynClient.Resource(corev1.SchemeGroupVersion.WithResource("secrets")).Namespace(namespace)
+				_, err := caSecretClient.Get(context.TODO(), CASecretName, metav1.GetOptions{})
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						logger.V(1).Info("CA secret not found during periodic check, skipping", "namespace", namespace)
+						continue
+					}
+					logger.Error(err, "Failed to check CA secret during periodic check", "namespace", namespace)
+					continue
+				}
+
+				// CA exists, check if it needs rotation
+				if err := s.IssueCA(context.TODO(), namespace); err != nil {
+					logger.Error(err, "Failed periodic CA check", "namespace", namespace)
+				}
+			case <-s.caStopCh:
+				return
+			}
+		}
+	}()
+
+	logger.Info("Started periodic CA check", "interval", CACheckInterval, "namespace", namespace)
+}
+
+func (s *signerImpl) stopPeriodicCACheck() {
+	if s.caTicker != nil {
+		s.caTicker.Stop()
+		s.caTicker = nil
+	}
+
+	if s.caStopCh != nil {
+		close(s.caStopCh)
+		s.caStopCh = nil
+	}
+
+	logger.Info("Stopped periodic CA check", "namespace", s.namespace)
+}
+
+// resignAllCSRs finds all existing CSR secrets and re-signs them with the new CA
+func (s *signerImpl) resignAllCSRs(ctx context.Context, namespace string) error {
+	secretClient := s.dynClient.Resource(corev1.SchemeGroupVersion.WithResource("secrets")).Namespace(namespace)
+
+	// List all secrets with the CSR request label
+	secretList, err := secretClient.List(ctx, metav1.ListOptions{
+		LabelSelector: SigningRequestLabelKey,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list CSR secrets: %w", err)
+	}
+
+	resignCount := 0
+	for _, item := range secretList.Items {
+		secret := &corev1.Secret{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(item.Object, secret); err != nil {
+			logger.Error(err, "Failed to convert secret", "secret", item.GetName())
+			continue
+		}
+
+		// Check if this secret has a CSR and was previously signed
+		if len(secret.Data[CSRDataKey]) == 0 {
+			continue
+		}
+
+		// Clear the signed annotation to trigger re-signing
+		if secret.Annotations == nil {
+			secret.Annotations = map[string]string{}
+		}
+		delete(secret.Annotations, RequestSignedLabelKey)
+
+		// Update the secret to trigger re-signing
+		secretUnstructured := resource.MustToUnstructured(secret)
+		if _, err := secretClient.Update(ctx, secretUnstructured, metav1.UpdateOptions{}); err != nil {
+			logger.Error(err, "Failed to update CSR secret for re-signing", "secret", secret.Name)
+			continue
+		}
```

Comment:
You can use `util.Update`:

```suggestion
		util.Update(ctx, resource.ForDynamic(secretClient), &item,
			func(existing *unstructured.Unstructured) (*unstructured.Unstructured, error) {
				annotations := existing.GetAnnotations()
				delete(annotations, RequestSignedLabelKey)
				existing.SetAnnotations(annotations)

				return existing, nil
			})
```